shortcircuit-api/src/main/java/com/powsybl/shortcircuit/Fault.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.shortcircuit;
+
+/**
+ * Interface to describe the characteristics of the fault to be simulated.
+ * Used for elementary short-circuit calculation only.
+ *
+ * @author Anne Tilloy <anne.tilloy at rte-france.com>
+ */
+public interface Fault {
```

Comment:
We need to add an enum for the type of fault (bus or branch):
this will be used to downcast, and to serialize/deserialize.

Reply:
> It would be better to have classes for faults: we will never want to have alternative implementations --> I thin we can promote the `AbstractFault` to `Fault` :)

After thinking again: not necessary after all.
Interface is fine as long as:
- we have an enum telling the type of the implementation
- we remove info specific to an implementation from the interface (proportional location)

If we go for a class, it will be a little strange since it should not be instantiable, and we will need to provide a dummy implementation for `getType` (or make it abstract, bu then we have a naming issue with checkstyle rules).